sdks/fabric/interoperation-node-sdk/src/MembershipManager.ts

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright IBM Corp. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+ 
+import { Gateway, Network } from 'fabric-network';
+import { Channel } from 'fabric-common';
+import * as path from 'path';
+import * as fs from 'fs';
+import { credentials } from '@grpc/grpc-js';
+
+
+import membership_pb from '@hyperledger-labs/weaver-protos-js/common/membership_pb';
+import agent_grpc_pb from '@hyperledger-labs/weaver-protos-js/identity/agent_grpc_pb';
+import agent_pb from '@hyperledger-labs/weaver-protos-js/identity/agent_pb';
+import common_ack_pb from "@hyperledger-labs/weaver-protos-js/common/ack_pb";
+
+import { handlePromise, promisifyAll } from './helpers'
+
+// Only Admin can create, update and delete local memberships
+async function createLocalMembership(
+    gateway: Gateway,
+    memberMspIds: Array<string>,
+    securityDomain: string,
+    channelName: string,
+    weaverCCId: string
+): Promise<any> {
+    const network = await gateway.getNetwork(channelName)
+    const membership = getMSPConfigurations(network, memberMspIds)
+    membership.setSecuritydomain(securityDomain)
+    const membership64 = Buffer.from(membership.serializeBinary()).toString('base64')
+    const contract = network.getContract(weaverCCId)
+    return await contract.submitTransaction("CreateLocalMembership", membership64);
+}
+
+async function updateLocalMembership(
+    gateway: Gateway,
+    memberMspIds: Array<string>,
+    securityDomain: string,
+    channelName: string,
+    weaverCCId: string
+): Promise<any> {
+    const network = await gateway.getNetwork(channelName)
+    const membership = getMSPConfigurations(network, memberMspIds)
+    membership.setSecuritydomain(securityDomain)
+    const membership64 = Buffer.from(membership.serializeBinary()).toString('base64')
+    const contract = network.getContract(weaverCCId)
+    return await contract.submitTransaction("UpdateLocalMembership", membership64);
+}
+
+async function deleteLocalMembership(
+    gateway: Gateway,
+    securityDomain: string,
+    channelName: string,
+    weaverCCId: string
+): Promise<any> {
+    const network = await gateway.getNetwork(channelName)
+    const contract = network.getContract(weaverCCId)
+    return await contract.submitTransaction("DeleteLocalMembership", securityDomain);
+}
+
+async function readLocalMembership(
+    gateway: Gateway,
+    securityDomain: string,
+    channelName: string,
+    weaverCCId: string
+): Promise<any> {
+    const network = await gateway.getNetwork(channelName)
+    const contract = network.getContract(weaverCCId)
+    return await contract.submitTransaction("DeleteLocalMembership", securityDomain);
+}
+
+function getMembershipUnit(channel: Channel, mspId: string): membership_pb.Member {
```

Comment:
Can we import this and the functions below into the IIN Agent rather than leave redundant code? We can do the same for the Fabric driver later.